util/retrieve-cluster-logs.py

```diff
@@ -0,0 +1,213 @@
+    return "{cluster_name}-logs-{end_time}".format(
+        cluster_name=cluster_name, end_time=datetime.now().strftime("%Y%m%d%H%M%S")
+    )
+
+
+def get_log_group(args):
+    """Get log group for args.cluster."""
+    log_group_name = "/aws/parallelcluster/{}".format(args.cluster)
+    logs = boto3.client("logs", region_name=args.region)
+    paginator = logs.get_paginator("describe_log_groups")
+    for result in paginator.paginate(logGroupNamePrefix=log_group_name):
+        for group in result.get("logGroups"):
+            if group.get("logGroupName") == log_group_name:
+                return group
+    err_and_exit(
+        "Unable to find log group for cluster {cluster}. Expected to find one named {log_group_name}".format(
+            cluster=args.cluster, log_group_name=log_group_name
+        )
+    )
+
+
+def verify_bucket_exists_in_region(args):
+    """Verify that bucket exists."""
+    bucket = boto3.resource("s3", region_name=args.region).Bucket(args.bucket)
+    if bucket.creation_date is None:
+        err_and_exit("s3 bucket {} does not exist.".format(args.bucket))
+    bucket_region = bucket.meta.client.get_bucket_location(Bucket=bucket.name).get("LocationConstraint")
+    if bucket_region != args.region:
+        err_and_exit(
+            "CloudWatch requires buckets to use for exporting logs to be in the same region as the log group. "
+            "The given cluster's log group is in {region}, but the given bucket's region is {bucket_region} ".format(
+                region=args.region, bucket_region=bucket_region
+            )
+        )
+
+
+def parse_args():
+    """Parse command line args."""
+    parser = argparse.ArgumentParser(description="Create an archive for a ParallelCluster's CloudWatch logs.")
+    parser.add_argument("--bucket", required=True, help="s3 bucket to export CloudWatch logs data to.")
+    parser.add_argument("--cluster", required=True, help="Name of cluster whose logs to get.")
+    parser.add_argument(
+        "--bucket-prefix", help="Keypath under which exported CloudWatch logs data will be stored in s3 bucket."
+    )
+    parser.add_argument("--region", required=True, help="Region in which the CloudWatch log group exists.")
+    args = parser.parse_args()
+    if not args.bucket_prefix:
+        args.bucket_prefix = get_default_bucket_prefix(args.cluster)
+    args.log_group = get_log_group(args)
+    verify_bucket_exists_in_region(args)
+    return args
+
+
+def start_export_task(logs_client, args):
+    """Start the task that will export the cluster's logs to an s3 bucket, and return the task ID."""
+    LOGGER.info(
+        "Starting export of logs from log group {log_group} to s3 bucket {bucket}".format(
+            log_group=args.log_group.get("logGroupName"), bucket=args.bucket
+        )
+    )
+    start_time = datetime.fromtimestamp(args.log_group.get("creationTime") / 1000)
+    end_time = datetime.now()
```

Comment:
same here for utc and everywhere else we are using a timestamp which depends on the timezone

Reply:
I thought the same thing, but it works to pass the start and end times around as timezone unaware `datetime`s and then convert them to a timestamp as they are passed into `create_export_task`.

The `datetime`'s `timestamp` method adjusts its return value based on the instance's timezone information, so passing around the UTC versions of the objects without explicitly attaching that timezone information results in times that are 9 hours later than intended when run from a PST locale.